app/services/unwithdraw_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class UnwithdrawService
+  def process(edition, user = nil)
+    Document.transaction do
+      edition.document.lock!
+      check_publishable(edition)
+
+      published_status = edition.status.details.published_status
+
+      edition.assign_status(published_status.state, user)
+      edition.save!
+
+      TimelineEntry.create_for_status_change(
+        entry_type: :unwithdrawn,
+        status: edition.status,
+      )
+
+      GdsApi.publishing_api_v2.republish(
+        edition.content_id,
```

Comment:
We should bung locale in here too for consistency with other Publishing API calls.